integration-tests/test_registration.py

```diff
@@ -93,3 +98,43 @@ def test_register_group_option(insights_client, legacy_upload_value):
         check=False,
     )
     assert register_group_option.returncode == 0
+
+
+def test_machineid_changes_when_insights_auth_method_changes(
+    insights_client, subman, test_config
+):
+    """machine-id should be different based on the auth method"""
+    if "satellite" in test_config.environment:
+        pytest.skip("this test requires BASIC authentication")
+
+    assert not insights_client.is_registered
+    insights_client.register()
+    with open(MACHINE_ID_FILE, "r") as f:
+        machine_id_cert = f.read()
+    insights_client.unregister()
+
+    insights_client.config.auto_config = False
+    insights_client.config.authmethod = "BASIC"
+    insights_client.config.username = test_config.get("candlepin", "username")
+    insights_client.config.password = test_config.get("candlepin", "password")
+    insights_client.config.save()
```

Comment:
After some testing, this doesn't look like a good way of doing it, because automatic configuration also configured prod/stage access.

Instead, I think we'll need to unregister and register sub-man here.

```py
try:
    subman.unregister()
    
    insights_client.register()
    ...
finally:
    subman.register()

assert ...
```